lib/active_merchant/billing/gateways/checkout_v2.rb

```diff
@@ -459,46 +457,50 @@ def access_token_url
       end
 
       def setup_access_token
-        request = 'grant_type=client_credentials'
-        begin
-          raw_response = ssl_post(access_token_url, request, access_token_header)
-        rescue ResponseError => e
-          raise OAuthResponseError.new(e)
-        else
-          response = parse(raw_response)
+        return if @options[:secret_key] || @options[:access_token]
 
-          if (access_token = response['access_token'])
-            access_token
-          else
-            raise OAuthResponseError.new(response)
-          end
-        end
+        response = parse(ssl_post(access_token_url, 'grant_type=client_credentials', access_token_header))
+        @options[:access_token] = response['access_token']
+      rescue ResponseError => e
+        @multiresponse = MultiResponse.new
```

Comment:
Why add a MultiResponse here?

Reply:
The initial idea was to obtain the response when the `access_token` is created so that we can debug better in case something fails and know what is related to the `access_token`, I moved this logic elsewhere and I think it looks better :)